### ISSUE TITLE
chezmoi is not respecting my .chezmoiignore

### ISSUE THREAD

**Issue by yusing** [twpayne/chezmoi#3630]
## Describe the bug

My .chezmoiignore inside subdirectory ~/docker/compose

```console
> cat .chezmoiignore
───────┬────────────────────────────────────────
       │ File: .chezmoiignore
───────┼────────────────────────────────────────
   1   │ data/
   2   │ data/**
   3   │ certs/
   4   │ certs/**
───────┴────────────────────────────────────────
```

```console
> chez add ~/docker
chezmoi: lstat /home/yusing/docker/compose/data/alist/opt/alist/data/log/log.log: permission denied
```

## To reproduce

`chez add ~/docker`

## Expected behavior

No output and error

## Output of command with the `--verbose` flag

```console
> chez --verbose add ~/docker
chezmoi: lstat /home/yusing/docker/compose/data/alist/opt/alist/data/log/log.log: permission denied
```

## Output of `chezmoi doctor`

<details>

```console
> chez doctor
RESULT    CHECK                       MESSAGE
ok        version                     v2.47.1, commit 1ce6b2eeb0caf75bd91883e5a968e713a26e7be2, built at 2024-03-03T00:47:52Z, built by Homebrew
ok        latest-version              v2.47.1
ok        os-arch                     linux/amd64 (Pop!_OS 22.04 LTS)
ok        uname                       Linux homelab 6.6.19-rt24-x64v4-xanmod1 #0~20240303.gf03df4f SMP PREEMPT_RT Sun Mar  3 12:38:55 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
ok        go-version                  go1.22.0 (gc)
ok        executable                  /home/linuxbrew/.linuxbrew/Cellar/chezmoi/2.47.1/bin/chezmoi
ok        upgrade-method              brew-upgrade
ok        config-file                 no config file found
warning   source-dir                  ~/.local/share/chezmoi is a git working tree (dirty)
ok        suspicious-entries          no suspicious entries
warning   working-tree                ~/.local/share/chezmoi is a git working tree (dirty)
ok        dest-dir                    ~ is a directory
ok        umask                       002
ok        cd-command                  found /usr/bin/zsh
ok        cd-args                     /usr/bin/zsh
info      diff-command                not set
ok        edit-command                found /home/linuxbrew/.linuxbrew/bin/nvim
ok        edit-args                   /home/linuxbrew/.linuxbrew/bin/nvim
ok        git-command                 found /usr/bin/git, version 2.34.1
warning   merge-command               vimdiff not found in $PATH
ok        shell-command               found /usr/bin/zsh
ok        shell-args                  /usr/bin/zsh
info      age-command                 age not found in $PATH
ok        gpg-command                 found /usr/bin/gpg, version 2.2.27
info      pinentry-command            not set
info      1password-command           op not found in $PATH
info      bitwarden-command           bw not found in $PATH
info      bitwarden-secrets-command   bws not found in $PATH
info      dashlane-command            dcli not found in $PATH
info      doppler-command             doppler not found in $PATH
info      gopass-command              gopass not found in $PATH
info      keepassxc-command           keepassxc-cli not found in $PATH
info      keepassxc-db                not set
info      keeper-command              keeper not found in $PATH
info      lastpass-command            lpass not found in $PATH
ok        pass-command                found /usr/bin/pass, version 1.7.4
info      passhole-command            ph not found in $PATH
info      rbw-command                 rbw not found in $PATH
info      vault-command               vault not found in $PATH
info      vlt-command                 vlt not found in $PATH
info      secret-command              not set
```

</details>

**Comment by twpayne** [comment]
What is the output of

```console
$ ls -l /home/yusing/docker/compose/data/alist/opt/alist/data/log
```

?

**Comment by twpayne** [comment]
For what it's worth, I suspect that this is the same underlying issue as #3588 in that you have files that in your home directory that you're not allowed to read and chezmoi is not handling this case correctly.

**Comment by yusing** [comment]
> What is the output of
> 
> ```
> $ ls -l /home/yusing/docker/compose/data/alist/opt/alist/data/log
> ```
> 
> ?

```console
sudo ls -l /home/yusing/docker/compose/data/alist/opt/alist/data/log
[sudo] password for yusing:
total 2796
-rw-r--r-- 1 root root 2858552 Mar  7 05:15 log.log
```

What I am thinking is, it should not even walk through /home/yusing/docker/compose/data since it is excluded in `.chezmoiignore`

**Comment by twpayne** [comment]
> What I am thinking is, it should not even walk through /home/yusing/docker/compose/data since it is excluded in `.chezmoiignore`

Yup, agree.

**Comment by twpayne** [comment]
Thanks for reporting this. This is now fixed with #3673.
